### ISSUE TITLE
"Parser cycle without extracting any bytes" warning should not be printed for parser cycle that advances the packet pointer

### ISSUE THREAD

**Issue by kfcripps** [p4lang/p4c#3544]
I noticed that when I build test testdata/p4_16_samples/parser-unroll-issue3537-1.p4, the following warning is emitted:
```
[--Wwarn=invalid] warning: Parser cycle without extracting any bytes:
Parser ParserI state chain: start, s1, s2, s2
```
Why is this warning printed here? Why do I care that I have a parser cycle that doesn't extract any bytes? In the case of this example, I only want to advance the packet cursor without extracting any headers, so I'm not sure what I can do to get rid of this warning.

_Originally posted by @kfcripps in https://github.com/p4lang/p4c/issues/3538#issuecomment-1264031764_

**Comment by kfcripps** [comment]
My main concern is not that the warning is printed, but rather that
> the "advance" call is not accounted in the code that checks whether the packet pointer has been advanced

which could result in other (currently unknown) unintended behavior in the `ParsersUnroll` pass

**Comment by mbudiu-vmw** [comment]
Fixed by #3548
